File: cpp_solutions/1051.cpp
class Solution {
public:
    int heightChecker(vector<int>& heights) {
        vector<int> orderedHeights(heights.begin(), heights.end());
        sort(heights.begin(), heights.end());
        int res = 0 ;
        int n = heights.size();
        for (int i = 0; i < n; ++i) {
            if (heights[i] != orderedHeights[i]) res++;
        }
        return res;
    }
};

File: rust_solutions/1051.rs
impl Solution {
    pub fn height_checker(heights: Vec<i32>) -> i32 {
        let mut ordered_heights = heights.clone();
        ordered_heights.sort_unstable();
        let n = heights.len();
        let mut res = 0;
        for i in 0..n {
            if heights[i] != ordered_heights[i] {
                res += 1;
            }
        }
        res
    }
}
